packages/tempus/src/Tempus_StepperForwardEulerAppAction.hpp

```diff
@@ -0,0 +1,76 @@
+// @HEADER
+// ****************************************************************************
+//                Tempus: Copyright (2017) Sandia Corporation
+//
+// Distributed under BSD 3-clause license (See accompanying file Copyright.txt)
+// ****************************************************************************
+// @HEADER
+
+#ifndef Tempus_StepperForwardEulerAppAction_hpp
+#define Tempus_StepperForwardEulerAppAction_hpp
+
+#include "Tempus_config.hpp"
+#include "Tempus_SolutionHistory.hpp"
+#include "Tempus_StepperForwardEuler.hpp"
+
+
+namespace Tempus {
+
+// Forward Declaration for recursive includes (this AppAction <--> Stepper)
+template<class Scalar> class StepperForwardEuler;
+
+/** \brief Application Action for StepperForwardEuler.
+ *
+ *  This class provides a means to apply various actions with the ForwardEuler time step.
+ *  The data available to this class is solution variables (through
+ *  SolutionHistory), and stepper data (through the Stepper).  It allows
+ *  the application to just observe this data (i.e., use but not change the
+ *  data) to change any of it (USER BEWARE!).
+ *
+ *  Below is the ForwardEuler algorithm and includes the locations where the
+ *  application can take actions (in italicized).
+ *
+ *  \f{algorithm}{
+ *  \renewcommand{\thealgorithm}{}
+ *  \caption{Forward Euler with the locations of the application actions indicated.}
+ *  \begin{algorithmic}[1]
+ *    \State Start with $x_n$, $\Delta t_n$
+ *    \State {\it appAction.execute(solutionHistory, stepper, BEGIN\_STEP)}
+ *    \State Form $f(x_{n},t_{n})$
+ *    \State {\it appAction.execute(solutionHistory, stepper, BEFORE\_EXPLICIT\_EVAL)}
+ *    \State Form $y \leftarrow x_{n} + \Delta t_n f(x_{n},t_n)$
+ *    \State {\it appAction.execute(solutionHistory, stepper, AFTER\_EXPLICIT\_EVAL)}
```

Comment:
Do we need AFTER_EXPLICIT_EVAL?  I did not add one for the RK methods, because there was no significant change to the solution between the explicit evaluation and the end stage.

Reply:
No, it's never used.  Should I go ahead and remove it?

Reply:
Yeah, I would so that it matches the other steppers better.